spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -822,20 +832,25 @@ class AbsHelperClass
     end
   end
 
-  # TODO: implement
   describe "#update_last_abs_resource_hosts" do
     context "when a host string is specified" do
       it "writes the string to the log file" do
-        skip
+        expect(File).to receive(:write).with("last_abs_resource_hosts.log", TEST_ABS_RESOURCE_HOSTS)
+        subject.update_last_abs_resource_hosts(TEST_ABS_RESOURCE_HOSTS)
       end
     end
   end
 
-  # TODO: implement
   describe "#backoff_sleep" do
     context "when called with a positive number of tries" do
-      it "sleeps for a multiple of the number of tries" do
-        skip
+      it "sleeps for a multiple of the number of tries and returns nil" do
+        tries = 2
+        sleep_time = 2**tries
+        message = "Sleeping for #{sleep_time} seconds after attempt #{tries}..."
+        expect(subject).to receive(:sleep).with(2**tries)
+        expect(subject).to receive(:puts).with(message)
```

Comment:
Is the STDOUT message a crucial component to this method that needs to be tested?

Reply:
It is the feedback that lets the user know what is happening during the host verification process so I wanted to ensure that the values are correct. The exact wording isn't crucial but also probably won't change so it seemed simplest to verify the entire message.